library/src/morpheus-master/sass/themes/_base.scss

```diff
@@ -1,49 +1,98 @@
-   // Color palette generation
-    // Use defineColorHSL to generate an 11 shade palette for a color.
-    // @include defineColorHSL(color, hue, saturation, lightness)
-    // @param color
-    //      Type: string
-    //      expects a CSS custom property name
-    // @param hue
-    //      Type: integer
-    //      expects an integer 0-255
-    // @param saturation
-    //      Type: percentage
-    //      expects a percentage 0-100%
-    // @param lightness
-    //      Type: percentage
-    //      expects a percentage 0-100%
-    // @returns 11 CSS Custom Properties available in :root
-    //      named color-0 through color-10
-    //      The HSL you provide in the parameters will be in the
-    //      middle of the range at -5.
-    // Example 
-    // @include defineColorHSL(--myschool-gray, 214, 3%, 55%);
-    //      returns --myschool-gray-5: hsl(214, 3%, 55%;
-    //
-    // Sakai comes predefined with gray, blue, lightblue, teal, gold,
-    // green, orange, purple, and red palettes available for you to use.
+
+// Color palette generation
+// Use defineColorHSL to generate an 11 shade palette for a color.
+// @include defineColorHSL(colorname, hue, saturation, lightness)
+// @param colorname
+//      Type: string
+//      expects a CSS custom property name
+// @param hue
+//      Type: integer
+//      expects an integer 0-255
+// @param saturation
+//      Type: percentage
+//      expects a percentage 0-100%
+// @param lightness
+//      Type: percentage
+//      expects a percentage 0-100%
+// @returns 11 CSS Custom Properties available in :root
+//      named colorname-0 through colorname-10
+//      The HSL you provide in the parameters will be in the
+//      middle of the range at colorname-5.
+// @returns 11 CSS Classes using colorname prefixed with sakai-colorize
+//      These classes set the background-color to colorname and
+//      set the color to a value which will always pass WCAG AA 
+//      guidelines. Color will be selected from the following,
+//      whichever first passes the contrast requirement.
+//            colorname-1
+//            colorname-9
+//            white
+//            black
+//
+// Example usage 
+//      @include defineColorHSL(--myschool-gray, 214, 3%, 55%);
+// Returns CSS Custom Properties on :root
 @mixin defineColorHSL($color, $hue, $saturation, $lightness){
```

Comment:
refactor to more programmatically create the color palettes. also generates a class available on :root for each color in the palette that defines a guaranteed to be accessible color/background color combination.